src/app/home/home.component.spec.ts

```diff
@@ -1,34 +1,146 @@
+    let restaurant_debug: DebugElement;
+    let restaurant_html:  HTMLInputElement;
+    let area_debug:       DebugElement;
+    let area_htlml:       HTMLInputElement;
+    let search_debug:     DebugElement;
+    let search_html:      HTMLElement;
+
+    beforeEach(async(() => {
+      
+      TestBed.configureTestingModule({
+        declarations: [ HomeComponent ],
+        imports: [
+          HttpModule, 
+          FormsModule,
+          MatAutocompleteModule,
+          MatToolbarModule,
+          MatFormFieldModule,
+          MatCardModule,
+          MatButtonModule,
+          MatSelectModule,
+          ReactiveFormsModule
+        ],
+        providers:[{
+          provide: ComponentFixtureAutoDetect, useValue:true
+        }
+
+        ]
+      })
+      .compileComponents();// declare the test component
+      }));
+  
+      beforeEach(() =>{
+          fixture=TestBed.createComponent(HomeComponent);
+          comp = fixture.componentInstance;   
+          fixture.detectChanges();
+          title_debug = fixture.debugElement.query(By.css('h1'));
+          const input = fixture.debugElement.queryAll(By.css('input'));
+          search_debug = fixture.debugElement.query(By.css('button'));
+          search_html= search_debug.nativeElement;
+          title_html = title_debug.nativeElement;
+          restaurant_html = input[0].nativeElement;
+          area_htlml = input[1].nativeElement;
+    });
+
+    it('should display original title', () => {
+    
+      expect(title_html.textContent).toContain(comp.title);
+    });
+    
+    it('should display a different test title', () => {
+      comp.title = 'Test Title';
+      fixture.detectChanges();
+      expect(title_html.textContent).toContain('Test Title');
+    });
+
+    it('checks home input ', () => {
+      expect(restaurant_html.textContent).toEqual('');
+      
+    });
+
+   it('cheks button', async(() => {
+     searchspy = spyOn(comp, 'SearchRestaurants').and.callThrough();
+     const searchBtn = fixture.debugElement.nativeElement.querySelector('button');
+       searchBtn.click();
+       fixture.whenStable().then(()=>{
+        expect(comp.SearchRestaurants).toHaveBeenCalled() 
+      });      
+     }));
+  
+  //  it('cheks button with values', async(() => {
+  //   searchspy = spyOn(comp, 'SearchRestaurants');
+  //   const searchBtn = fixture.debugElement.nativeElement.querySelector('button');
+ 
+  //   searchBtn.click();
+    
+  //   fixture.detectChanges();
+  //   // expect($('#sandbox')).toHaveValue(value);
+  //   restaurant_html.defaultValue = 'burger';
+  //   restaurant_html.dispatchEvent(new Event('input'));
+  //   expect(restaurant_html.defaultValue).toEqual("burger"); 
+  //   expect(comp.SearchRestaurants).toHaveBeenCalledWith('burger','london'); 
+  //   fixture.whenStable().then(()=>{
+  //   expect(comp.SearchRestaurants).toHaveBeenCalledWith(restaurant_html.textContent,area_htlml.textContent); 
+  //    });      
+  //   }));
```

Comment:
Can you check whether to remove last test case

Reply:
test case is commented for future reference.